test/unit/coffee/ChannelManagerTests.coffee

```diff
@@ -17,104 +17,155 @@ describe 'ChannelManager', ->
 	describe "subscribe", ->
 
 		describe "when there is no existing subscription for this redis client", ->
-			beforeEach ->
+			beforeEach (done) ->
 				@rclient.subscribe = sinon.stub().resolves()
 				@ChannelManager.subscribe @rclient, "applied-ops", "1234567890abcdef"
+				setTimeout done
 
 			it "should subscribe to the redis channel", ->
 				@rclient.subscribe.calledWithExactly("applied-ops:1234567890abcdef").should.equal true
 
 		describe "when there is an existing subscription for this redis client", ->
-			beforeEach ->
+			beforeEach (done) ->
 				@rclient.subscribe = sinon.stub().resolves()
 				@ChannelManager.subscribe @rclient, "applied-ops", "1234567890abcdef"
-				@rclient.subscribe = sinon.stub().resolves()  # discard the original stub
 				@ChannelManager.subscribe @rclient, "applied-ops", "1234567890abcdef"
+				setTimeout done
 
 			it "should not subscribe to the redis channel", ->
-				@rclient.subscribe.called.should.equal false
+				@rclient.subscribe.callCount.should.equal 1
```

Comment:
Replacing the stub causes a race condition with the _always delayed_ `actualSubscribe` execution. `rclient.subscribe` is a single stub instance now. There is a similar change right below.